### PR TITLE
feat: Quasiquotation

### PR DESCRIPTION
**Pull request by hellerve**
This PR adds quasiquotation to the core. It’s implemented entirely in Carp, as a macro.

Test cases and documentation are included.

Cheers

### CODE REVIEW

File: core/Quasiquote.carp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+```
+(defdynamic x 1)
+
+(quasiquote (+ (unquote x) 1)) ; => (+ 1 1)
+```")
+  (defmacro unquote [arg]
+    (macro-error "unquotes need to be wrapped inside quasiquotes."))
+
+  (doc unquote "unquotes (i.e. evaluates) an expression `arg` inside a
```
Comment (jacereda) [inline core/Quasiquote.carp:20]: You probably mean `doc unquote-splicing` here.

Comment (hellerve) [reply core/Quasiquote.carp]: D'oh! Good catch, thank you!

**Review by jacereda — APPROVED** [review]

**Comment by hellerve** [comment]
I just noticed a problem (and fixed it) with the evaluation schema that was there before, because it was too eager (it should rewrite instead of evaluate, since we might use it inside dynamic functions).

File: core/Quasiquote.carp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+quasiquote. It will generate an error message when used outside a
+`quasiquote` form.
+
+Example:
+
+```
+(defdynamic x 1)
+
+(quasiquote (+ (unquote x) 1)) ; => (+ 1 1)
```
Comment (eriksvedang) [inline core/Quasiquote.carp:15]: Maybe use two different numbers here to make it more obvious what happens?

File: docs/Quasiquotation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Quasiquotation is a way to quote parts of a list while evaluating others. It’s
+only available in the dynamic parts of the program.
+
+Quasiquotation enables the user to intersperse evaluated or “unquoted” portions
+into a unevaluated or “quoted” list.
+
+```clojure
+(defdynamic x 1)
+
+(quasiquote (+ (unquote x) 1)) ; => (+ 1 1)
```
Comment (eriksvedang) [inline docs/Quasiquotation.md:12]: Same here.

**Review by eriksvedang — APPROVED** [review]

**Pull request merged**